spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -437,6 +433,24 @@ protected void actualStart() throws Exception {
 		}
 	}
 
+	protected void doRedeclareElementsIfNecessary() {
+		String routingLookupKey = getRoutingLookupKey();
+		if (routingLookupKey != null) {
+			SimpleResourceHolder.bind(getRoutingConnectionFactory(), routingLookupKey);
```

Comment:
Well, actually only `SimpleRoutingConnectionFactory` deals with the `SimpleResourceHolder` maybe that is a bit expensive to store to `ThreadLocal` always unconditionally ?